content/en/platform/corda/4.11/enterprise/two-phase-finality.md

```diff
@@ -0,0 +1,69 @@
+---
+date: '2023-03-30T12:00:00Z'
+menu:
+  corda-enterprise-4-11:
+    parent: corda-enterprise-4-11-cordapps-flows
+tags:
+- two
+- phase
+- finality
+title: Two Phase Finality
+weight: 40
+---
+
+## Introduction
+
+Finality refers to the act of notarizing, recording, and sharing a transaction with all of its participants. Finality enables ledger consistency.
+
+An **initiator** uses the built-in flow called `FinalityFlow` to finalize a transaction:
+
+1. Send the transaction to the chosen notary and, if necessary, satisfy the notary that the transaction is valid.
+2. Record the transaction in the local vault, if it is relevant (that is, it involves the owner of the node).
+3. Send the fully signed transaction to the other participants for recording also.
+
+One or more **receivers** use the built-in flow `ReceiveFinalityFlow` to receive and record the finalized transaction.
+
+## Conventional implementation
+
+Up until Corda 4.10, the finality flow protocol was implemented using a single pass transaction sharing mechanism as depicted below:
+
+{{< figure alt="Conventional Single Phase Finality Protocol" width=100% zoom="./resources/C4-finality-conventional.png" >}}
+
+{{< note >}}
+* Peers only receive the finalized transaction after successful notarization.
+* Recovery is only possible at the initiator side, as peers have no record of a transaction until finality.
+* Failure conditions may lead to ledger inconsistency, requiring manual intervention for recovery.
+{{< /note >}}
+
+## Two Phase Finality (2PF)
+
+To address the shortcomings of the conventional protocol, Two Phase Finality introduces a multi-phased protocol whereby:
+
+* All parties have a copy of the unnotarized transaction. (Phase 1)
+* Additional metadata is stored with the unnotarized transaction to aid recovery.
+* All parties eventually have a copy of the notarized transaction. (Phase 2)
+* Recovery is possible at both the initiator and the receiver sides.
+* Failure conditions may lead to ledger inconsistency, which is recoverable using a new suite of
+  [finality flow recovery tools and commands]({{< relref "finality-flow-recovery.md" >}}).
+
+The following diagram illustrates the Two Phase Finality protocol:
+
+{{< figure alt="Two Phase Finality Protocol" width=100% zoom="./resources/C4-finality-optimized-two-phase-finality.png" >}}
+
+The two primary optimizations used within the protocol are:
+
+* Usage of a *Deferred Acknowledgment* in Phase 1, where the Receiver sends back an explicit `FetchDataFlow.Request.End`
+  acknowledgement to the initiator `SendTransaction` flow.
+
+  Note that the `ReceiverTransactionFlow` is now passed an optional parameter (`deferredAck` = true) to instruct it to not perform any final acknowledging.

```

Comment:
```suggestion
  Note that `ReceiverTransactionFlow` is now passed an optional parameter (`deferredAck` = true) to instruct it to not perform any final acknowledging.

```

Reply:
I think either of these work:

- Note that `ReceiverTransactionFlow` is now...
- 
- Note that the `ReceiverTransactionFlow` parameter is now ....

But not the above - what do you think?